core/src/main/java/org/geysermc/geyser/pack/ResourcePack.java

```diff
@@ -97,6 +104,11 @@ public static void loadPacks() {
                             }
                         }
                     });
+
+                    // Check if a file exists with the same name as the resource pack suffixed by .key,
+                    // and set this as content key. (e.g. test.zip, key file would be test.zip.key)
+                    File keyFile = new File(file.getParentFile(), file.getName() + ".key");
```

Comment:
Pretty sure you can use `directory` instead of `file.getParentFile()`

Reply:
Technically yes as it's not a walking directory search, but this would be always valid ^^